channels/rdpdr/client/rdpdr_main.c

```diff
@@ -443,16 +487,23 @@ static void* drive_hotplug_thread_func(void* arg)
 	fd_set rfds;
 	struct timeval tv;
 	int rv;
+	WIN32ERROR error;
 
 	rdpdr = (rdpdrPlugin*) arg;
 
-	rdpdr->stopEvent = CreateEvent(NULL, TRUE, FALSE, NULL);
+	if (!(rdpdr->stopEvent = CreateEvent(NULL, TRUE, FALSE, NULL)))
+	{
+		WLog_ERR(TAG, "CreateEvent failed!");
+		ExitThread(ERROR_INTERNAL_ERROR);
+		return NULL;
+	}
 
 	mfd = open("/proc/mounts", O_RDONLY, 0);
 
 	if (mfd < 0)
 	{
 		WLog_ERR(TAG, "ERROR: Unable to open /proc/mounts.");
+		ExitThread(ERROR_INTERNAL_ERROR);
```

Comment:
Maybe I've missed it somehow but shouldn't we inform the main thread if we exit?